TYPO3.Flow/Classes/TYPO3/Flow/Aop/Builder/ProxyClassBuilder.php

```diff
@@ -540,8 +540,21 @@ protected function addBuildMethodsAndAdvicesCodeToClass($className, ClassNameInd
         }
 
         $callBuildMethodsAndAdvicesArrayCode = "        if (method_exists(get_parent_class(), 'Flow_Aop_Proxy_buildMethodsAndAdvicesArray') && is_callable('parent::Flow_Aop_Proxy_buildMethodsAndAdvicesArray')) parent::Flow_Aop_Proxy_buildMethodsAndAdvicesArray();\n";
-        $proxyClass->getConstructor()->addPreParentCallCode($callBuildMethodsAndAdvicesArrayCode);
-        $proxyClass->getMethod('__wakeup')->addPreParentCallCode($callBuildMethodsAndAdvicesArrayCode);
+
+        $proxyMethods = ['__construct' => null, '__wakeup' => null];
+        if ($this->reflectionService->isMethodFinal($className, '__construct') === false) {
+            $proxyClass->getConstructor()->addPreParentCallCode($callBuildMethodsAndAdvicesArrayCode);
+            $proxyMethods['__construct'] = true;
+        }
+
+        if ($this->reflectionService->isMethodFinal($className, '__wakeup') === false) {
+            $proxyClass->getMethod('__wakeup')->addPreParentCallCode($callBuildMethodsAndAdvicesArrayCode);
+            $proxyMethods['__wakeup'] = true;
+        }
+
+        foreach (array_keys(array_filter($proxyMethods, 'is_null')) as $methodName) {
```

Comment:
Can't because I need the "other" value, not the true value. doing array_filter without anything would give me the methods that were proxied. I could name the variable negative, initialize to true and set to false if a proxy was created, but that felt like double negative. This is the most concise way to do this I could come up with.